### DISCUSSION TITLE
How to follow wikilinks under cursor?

### DISCUSSION THREAD

**Discussion by prmadev** [zk-org/zk-nvim#17] [Help] [answered]
Hello! Thanks for all your work! It really is a an amazing piece of work!
I'm trying to figure out the workflow. 
Is there a way to open a link under cursor? ``<cr>`` doesn't work, and ``gf`` only works if the I add ``.md`` to the wikilink. I tried a few plugins, but most of them only work with markdown format. Is there a trick or should I just use ``:ZkLinks``?

**Comment by mickael-menu** [comment]
Hello,

This should work.
Could you share your `.zk/config.toml` file, as well as an example of a link that doesn't work, and the path (relative to notebook root) + title of the note you expect should open?

**Comment by karimknaebel** [comment, marked as answer]
Following links works via LSP "go to definition".

The following should work `:lua vim.lsp.buf.definition()`. You've probably already bound this to something like `gd` if you use LSP.

**Reply by mickael-menu** [reply]
Ha right, I assumed `gf` was bound to this but it's just the default "go to file" Vim binding.

**Reply by prmadev** [reply]
Thank you both! that worked, and I now have a better understanding of the mechanic of the plugin! Thanks!

**Reply by karimknaebel** [reply]
You're welcome. And just FYI, this is not the only LSP backed functionality that you can use. For example you can `vim.lsp.buf.hover()` on a link to show a preview of the note in a popup window.

See https://github.com/mickael-menu/zk/issues/22 for what else is supported :)

**Reply by mickael-menu** [reply]
@amirography I opened a PR with more mapping examples: https://github.com/mickael-menu/zk-nvim/pull/18

**Reply by prmadev** [reply]
> @amirography I opened a PR with more mapping examples: #18

Thanks! I really appreciate it!

**Comment by jfvillablanca** [comment]
Hello, I'm having troubleshooting with `gd`. `gd` used to work. After some hiatus of maintaining my notes, I returned to reorganize but now, I'm unable to `gd` to other files. I just updated zk via brew but the problem still persists.

I have lsp installed and checking LspInfo, zk seems to be attached on the buffer. What could be the cause of such a problem so I may investigate it?

**Reply by mickael-menu** [reply]
It's probably not mapped to `vim.lsp.buf.definition()` in your config (it's not done by default).

Take a look at the mapping examples in the README, and you can check your current mapping with `:map`.

Personally I'm using `<CR>` for that and have:
<img width="572" alt="Screenshot 2023-01-22 at 09 35 02" src="https://user-images.githubusercontent.com/58686775/213907252-04768a98-9731-4aa7-b578-2f0ee601f785.png">

**Reply by jfvillablanca** [reply]
I have it mapped too. I use `gd` to navigate my nvim config and it works.
![Screenshot from 2023-01-23 08-03-08](https://user-images.githubusercontent.com/31008330/213947637-c9c7ae75-e311-41ff-828c-8a7a5ce04219.png)

One thing I noticed is, on some notes, when I `gd` the cursor moves to the top. However, I don't know how to replicate it. Still investigating why

**Reply by jfvillablanca** [reply]
I discovered something, so I think I have narrowed down on the issue. Yesterday, I uninstalled `grammarly-languageserver`, that's when `gd` stopped working.
I reinstalled it and `gd` works now. 
I know it's not only `gd` since all LSP remaps does not work (i checked Telescope keymaps). 

Checking LspInfo on a zk note buffer, I have 3 servers attached to the buffer:
`grammarly`'s autostart is set to `true`;
`zk` and `null-ls` are set to `false`
It's odd that some language server has an effect on another language server, so it must be due to servers attaching to the buffer.

Is it possible to uninstall `grammarly-language-server` while maintaining LSP functionality in the buffer? Seems like an odd interaction

**Reply by eric-hansen** [reply]
You can't really maintain LSP functionality if the LSP client/server isn't communicating.

Unless there's a reason for the Grammarly LSP to be connectd to the zk buffers, why not have it ignore the zk note buffers (i.e.: via filetype) and have the zk LSP active on said buffers?
